node/src/store/sled_store.rs

```diff
@@ -210,6 +236,60 @@ impl SledStore {
         Ok(())
     }
 
+    async fn mark_header_sampled(
+        &self,
+        height: u64,
+        accepted: bool,
+        cids: Vec<Cid>,
+    ) -> Result<u64> {
+        let inner = self.inner.clone();
+
+        spawn_blocking(move || {
+            // Make sure we have the header being marked
+            let head_height = read_height_by_db_key(&inner.db, HEAD_HEIGHT_KEY).unwrap_or(0);
```

Comment:
I think there is a funny corner case when someone could insert sampling metadata for height 0 if store didn't have any header appended yet

```suggestion
            let head_height = read_height_by_db_key(&inner.db, HEAD_HEIGHT_KEY)?;
```

Reply:
You're right, it also looks like `contains_height` doesn't always check for 0 correctly.
Fixed both above issues and added unittests for 0-height headers and sampling.